app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,52 @@
+class CategoriesController < ApplicationController 
+
+    rescue_from ActiveRecord::RecordNotFound, with: :category_not_found
+    # GET all the categories
+    def index 
+        categories = Category.all 
+        render json: categories, status: :ok
+    end 
+    # GET one category 
+    def show 
+        category = find_category
+        render json: category, status: :ok
+    end 
+
+    # POST a new category
+    def create 
+        category = Category.create(category_params)
+        render json: category, status: :created
+    end
+
+    # PATCH a category 
+    def update 
+        category = find_category
+        category.update(category_params)
+        render json: category, status: :accepted
+    end 
+
+    # DELETE a category
+    def destroy 
+        category = find_category
+        category.destroy
+        head :no_content
+    end  
+
+    # Private methods 
+    private 
```

Comment:
the private methods have simplified my code, thanks for the suggestion. Nice work here.